wordpress-objects/Clarkson_Post_Type.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * Clarkson Post Type.
+ *
+ * @package CLARKSON\Objects
+ */
+
+/**
+ * Clarkson Post Type class.
+ */
+class Clarkson_Post_Type {
+
+	/**
+	 * @var \WP_Post_Type
+	 */
+	protected $_post_type;
+
+	/**
+	 * @var array
+	 */
+	protected static $post_types;
+
+	public function __construct( \WP_Post_Type $post_type ) {
+		$this->_post_type = $post_type;
+	}
+
+	/**
+	 * Get post type object by post type name
+	 */
+	public static function get( string $post_type ): Clarkson_Post_Type {
+		$class            = get_called_class();
+		$post_type_object = get_post_type_object( $post_type );
+		return new $class( $post_type_object );
+	}
+
+	/**
+	 * Check if an existing field is requested.
+	 *
+	 * @param string $name Field to search by.
+	 *
+	 * @throws Exception Error message.
+	 */
+	public function __get( string $name ) {
+		if ( in_array( $name, array(), true ) ) {
+			throw new Exception( 'Trying to access wp_post_type object properties from Post Type object.' );
+		}
+	}
+
+	/**
+	 * Get the post type title.
+	 */
+	public function get_title(): string {
+		$title = $this->_post_type->labels->singular_name;
+		return apply_filters( 'the_title', $title );
```

Comment:
Wordpress had some filtering logic going on which i implemented.
It got lost during further development, but it needed to be this: https://developer.wordpress.org/reference/hooks/post_type_archive_title/

Magic get is also an idea i can try

Reply:
I've done it even a little bit different.
I've created a magical get to proxy unexisting properties to `WP_Post_Type` class.
I'm **not** proxying functions to `WP_Post_Type`, because they should never be called from our templates. Wordpress should make them private...

So you can now access the labels, and even more if required.
Also, the `get_title()` and `get_archive_title()` are still in place, returning the appropriate labels, with filters if required.